wp-content/mu-plugins/pantheon/pantheon-updates.php

```diff
@@ -19,6 +19,12 @@ function _pantheon_get_latest_wordpress_version() {
 		return null;
 	}
 
+	// We're only looking for stable releases so return null for unstable versions of WordPress, such as trunk or beta
+	// Stable releases only contain numbers and periods, e.g. 4.9
+	if( 0 === preg_match('/^[\d\.]+$/', $core_updates[0]->current) ){
```

Comment:
@westonruter how's this? Anything that doesn't contain a number or period will be marked as an unstable version and won't trigger the Pantheon update nag. I tested with standard versions like `4.8.2` and `4.9`.

Reply:
That works, though you could also just check to see if the version contains `-`.

Reply:
If I've installed `4.9-alpha-40870-src` and `4.9` comes out, will I receive an update nag?

Reply:
No, returning `null` here based on the current version not being a stable release will disable all update nags.

Reply:
> No, returning `null` here based on the current version not being a stable release will disable all update nags.

Hm. Personally, I'd expect to see an update nag if there's a stable release that's more recent than the development version I'm running. I often forget to keep development versions up to date :/

We could write a version comparison function that works with custom WP versions, and do so in a separate repo with tests, etc.

Frankly, WP-CLI would benefit from this too because Weston has just highlighted a couple of inadequacies to my existing helper function: https://github.com/wp-cli/wp-cli/blob/master/php/utils.php#L246

Reply:
https://github.com/wp-cli/wp-cli/issues/4380

Reply:
@danielbachhuber I think the comparison logic would be something like this:

```php
/**
 * Compare two WordPress version numbers which look like '4.9-alpha-40870-src'.
 *
 * @param string $ver_a Version A.
 * @param string $ver_b Version B.
 * @return int Comparison.
 */
$compare = function( $ver_a, $ver_b ) {
	$ver_a_parts = explode( '-', $ver_a );
	$ver_b_parts = explode( '-', $ver_b );

	// Compare semantic versions.
	$semver_a = array_shift( $ver_a_parts );
	$semver_b = array_shift( $ver_b_parts );
	$comparison = version_compare( $semver_a, $semver_b );
	if ( 0 !== $comparison ) {
		return $comparison;
	}

	// Compare pre-release identifiers 'alpha', 'beta', 'rc'.
	$prerelease_a = array_shift( $ver_a_parts );
	$prerelease_b = array_shift( $ver_b_parts );
	$comparison = strcmp( $prerelease_a, $prerelease_b );
	if ( 0 !== $comparison ) {
		return $comparison;
	}

	// If either version is a src (but not both) then consider them equal.
	$src_a = 'src' === array_shift( $ver_a_parts );
	$src_b = 'src' === array_shift( $ver_b_parts );
	if ( $src_a xor $src_b ) {
		return 0;
	}

	// Compare revision numers.
	$rev_a = intval( array_shift( $ver_a_parts ) );
	$rev_b = intval( array_shift( $ver_b_parts ) );
	return $rev_b - $rev_a;
};
```